src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/RecordComponentNumberCheck.java

```diff
@@ -0,0 +1,298 @@
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="RecordComponentNumber"/&gt;
+ * </pre>
+ * <p>
+ * To configure the check to allow 5 record components at all access modifier levels
+ * for record definitions:
+ * </p>
+ * <pre>
+ * &lt;module name="RecordComponentNumber"&gt;
+ *   &lt;property name="maxComponents" value="5"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Java code example:
+ * </p>
+ * <pre>
+ * public record MyRecord1(int x, int y, String str) { // ok, 3 components
+ *     ...
+ * }
+ *
+ * public record MyRecord2(int x, int y, String str,
+ *                           Node node, Order order, Data data) { // violation, 6 components
+ *     ...
+ * }
+ * </pre>
+ * <p>
+ * To configure the check to allow 10 record components for a public record definition,
+ * but 3 for private record definitions:
+ * </p>
+ * <pre>
+ * &lt;module name="RecordComponentNumber"&gt;
+ *   &lt;property name="maxComponents" value="3"/&gt;
+ *   &lt;property name="accessModifiers" value="private"/&gt;
+ * &lt;/module&gt;
+ * &lt;module name="RecordComponentNumber"&gt;
+ *   &lt;property name="maxComponents" value="10"/&gt;
+ *   &lt;property name="accessModifiers" value="public"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * Java code example:
+ * </p>
+ * <pre>
+ * public record MyRecord1(int x, int y, String str) { // ok, public record definition allowed 10
+ *     ...
+ * }
+ *
+ * private record MyRecord2(int x, int y, String str, Node node) { // violation
+ *     ...                                // private record definition allowed 3 components
+ * }
+ * </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code too.many.components}
+ * </li>
+ * </ul>
+ *
+ * @since 8.36
+ */
+@FileStatefulCheck
+public class RecordComponentNumberCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "too.many.components";
+
+    /** Default maximum number of allowed components. */
+    private static final int DEFAULT_MAX_COMPONENTS = 8;
+
+    /** Specify the maximum number of components allowed in record definition headers. */
+    private int maxComponents = DEFAULT_MAX_COMPONENTS;
+
+    /**
+     * Access modifiers of record definitions where the number
+     * of record components should be checked.
+     */
+    private AccessModifierOption[] accessModifiers = {
+        AccessModifierOption.PUBLIC,
+        AccessModifierOption.PROTECTED,
+        AccessModifierOption.PACKAGE,
+        AccessModifierOption.PRIVATE,
+    };
+
+    /**
+     * Setter to specify the maximum number of components allowed in record definition headers.
+     *
+     * @param value the maximum allowed.
+     */
+    public void setMaxComponents(int value) {
```

Comment:
Other properties like this just named `max`

Reply:
So, `setMax()`?

Reply:
Done.